### PR TITLE
feat(BPM failure): add BPM failures API resources

### PR DESCRIPTION
**Pull request by abirembaut**
Covers [BPM-316](https://bonitasoft.atlassian.net/browse/BPM-316)
Covers [BPM-328](https://bonitasoft.atlassian.net/browse/BPM-328)

[BPM-316]: https://bonitasoft.atlassian.net/browse/BPM-316?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[BPM-328]: https://bonitasoft.atlassian.net/browse/BPM-328?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: openapi/openapi.yaml [hunk 1]
```diff
@@ -551,6 +560,10 @@ paths:
     $ref: './paths/API@bpm@archivedFlowNode.yaml'
   /API/bpm/archivedFlowNode/{id}:
     $ref: './paths/API@bpm@archivedFlowNode@{id}.yaml'
+  /API/bpm/failure/flowNode{flowNodeInstanceId}:
```
Comment (rbioteau) [inline openapi/openapi.yaml:563]:
```suggestion
  /API/bpm/failure/flowNode/{flowNodeInstanceId}:
```

File: openapi/openapi.yaml [hunk 2]
```diff
@@ -551,6 +560,10 @@ paths:
...
+    $ref: './paths/API@bpm@failure@flowNode@{flowNodeInstanceId}.yaml'
+  /API/bpm/achivedFailure/flowNode{flowNodeInstanceId}:
```
Comment (rbioteau) [inline openapi/openapi.yaml:565]:
```suggestion
  /API/bpm/achivedFailure/flowNode/{flowNodeInstanceId}:
```

File: openapi/paths/API@bpm@archivedFailure@flowNode@{flowNodeInstanceId}.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  operationId: getArchivedBPMFailuresByFlowNodeInstanceId
+  parameters:
+    - description: The original identifier (sourceObjectID) of the flownode instance from which to retrieve the failures
+      in: path
+      name: flowNodeInstanceId
+      required: true
+      schema:
+        type: string
+        maxLength: 250
+        pattern: '^[A-Za-z0-9\_\-\.]{0,250}$'
```
Comment (rbioteau) [inline openapi/paths/API@bpm@archivedFailure@flowNode@{flowNodeInstanceId}.yaml:16]: Not sure A-Zaz is part of the pattern of a flownode id

File: openapi/paths/API@bpm@failure@flowNode@{flowNodeInstanceId}.yaml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  operationId: getBPMFailuresByFlowNodeInstanceId
+  parameters:
+    - description: The identifier of the flownode instance from which to retrieve the failures
+      in: path
+      name: flowNodeInstanceId
+      required: true
+      schema:
+        type: string
+        maxLength: 250
+        pattern: '^[A-Za-z0-9\_\-\.]{0,250}$'
```
Comment (rbioteau) [inline openapi/paths/API@bpm@failure@flowNode@{flowNodeInstanceId}.yaml:16]: Not sure A-Zaz is part of the pattern of a flownode id

**Review by rbioteau — CHANGES_REQUESTED** [review]

File: openapi/paths/API@bpm@archivedFailure@flowNode@{flowNodeInstanceId}.yaml (hunk 3 above)
Comment (abirembaut) [reply openapi/paths/API@bpm@archivedFailure@flowNode@{flowNodeInstanceId}.yaml:16]:
```suggestion
        pattern: '^[0-9\_\-\.]{0,250}$'
```

File: openapi/paths/API@bpm@failure@flowNode@{flowNodeInstanceId}.yaml (hunk 4 above)
Comment (abirembaut) [reply openapi/paths/API@bpm@failure@flowNode@{flowNodeInstanceId}.yaml:16]:
```suggestion
        pattern: '^[0-9\_\-\.]{0,250}$'
```

File: openapi/components/schemas/ArchivedBPMFailure.yaml [hunk 5]
```diff
@@ -0,0 +1,48 @@
+type: object
+description: "The archived version of a failure that happened during a BPM instance (flow node, process) execution that led to the instance being in a failed state."
+properties:
+  id:
```
Comment (rbioteau) [inline openapi/components/schemas/ArchivedBPMFailure.yaml:4]: Use alphabetical order

**Review by rbioteau — CHANGES_REQUESTED** [review]

**Review by rbioteau — CHANGES_REQUESTED** [review]

**Pull request merged**